harness/determined/estimator/_estimator_context.py

```diff
@@ -142,3 +144,26 @@ def allgather_metrics(self, metrics: Any) -> List:
         if self._allgather_fn is None:
             raise AssertionError("allgather_metrics must not be called before training begins")
         return self._allgather_fn(metrics)
+
+    def _build_allgather_op(self, f: Callable) -> Callable:
+        """Build an op that uses allgather in a way that is safely sequentialized."""
+
+        def decorated(*arg: Any, **kwarg: Any) -> tf.Operation:
+            with tf.compat.v1.control_dependencies(self._allgather_ops):
+                new_op = f(*arg, **kwarg)
+            self._allgather_ops.append(new_op)
+            return new_op
+
+        return decorated
+
+    def make_metric(
+        self,
+        metric: Any,
+        reducer: Union[Callable[[List[Any]], Any], "estimator.MetricReducer"],
```

Comment:
question: why is `estimator.MetricReducer` in quotes?

Reply:
It is a forward reference, like it is not yet defined when this portion of the code is read by the interpreter.